onnxruntime/contrib_ops/webgpu/quantization/matmul_nbits.cc

```diff
@@ -565,11 +827,51 @@ Status MatMulNBits::ComputeInternal(onnxruntime::webgpu::ComputeContext& context
   uint32_t components = GetMaxComponents(N);
 
   const bool has_zero_points = zero_points != nullptr;
+  const bool has_subgroup = context.Device().HasFeature(wgpu::FeatureName::Subgroups);
+  if (accuracy_level_ == 4 && block_size == 32 &&
```

Comment:
Also check whether the output type is `float16` or add the `float` support in the shader?

Reply:
added fp32 support, though not tested.